src/app/components/delete/DeleteModal.js

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { c, msgid } from 'ttag';
+import { useApi, useNotifications, useEventManager, useLoading, Alert, ErrorButton, FormModal } from 'react-components';
+
+import { clearContacts, deleteContacts } from 'proton-shared/lib/api/contacts';
+import { allSucceded } from 'proton-shared/lib/api/helpers/response';
+
+const DeleteModal = ({
+    activeIDs,
+    contactID,
+    contacts,
+    filteredContacts,
+    filteredCheckedIDs,
+    onCheck,
+    onClearSearch,
+    onUpdateChecked,
+    history,
+    location,
+    ...rest
+}) => {
+    const api = useApi();
+    const { createNotification } = useNotifications();
+    const { call } = useEventManager();
+    const [loadingDelete, withLoadingDelete] = useLoading();
+
+    const submit = <ErrorButton type="submit" loading={loadingDelete}>{c('Action').t`Delete`}</ErrorButton>;
+
+    const handleDelete = async () => {
+        if (activeIDs.length === contacts.length) {
+            await api(clearContacts());
+            history.replace({ ...location, pathname: '/contacts' });
+            await call();
+            onUpdateChecked(Object.create(null));
+            return createNotification({ text: c('Success').t`Contacts deleted` });
```

Comment:
Does the modal not need to be closed here too?

Reply:
oops